src/components/layout/CyVerseAppBar.js

```diff
@@ -122,18 +127,156 @@ function CyverseAppBar(props) {
             throwOnError: true,
         },
     });
-    React.useEffect(() => {
-        if (enabled && userProfile && userProfile.id) {
-            intercomLogin(
-                userProfile.id,
-                userProfile.attributes.email,
-                appId,
-                companyId,
-                companyName
+
+    const gotToOutputFolder = useCallback(
+        (outputFolder) => {
+            router.push(
+                `${constants.PATH_SEPARATOR}${NavigationConstants.DATA}${constants.PATH_SEPARATOR}ds${outputFolder}`
             );
+        },
+        [router]
+    );
+
+    const analysisCustomAction = useCallback(
+        (outputFolderPath) => {
+            return (
+                <Button
+                    variant="outlined"
+                    onClick={() => {
+                        console.log("output folder path=>" + outputFolderPath);
+                        gotToOutputFolder(outputFolderPath); //gotcha - I cannot do router.push from here
+                    }}
+                >
+                    <Typography
+                        variant="button"
+                        style={{ color: theme.palette.primary.contrastText }}
+                    >
+                        {formatMessage(intl, "viewOutput")}
+                    </Typography>
+                </Button>
+            );
+        },
+        [gotToOutputFolder, intl, theme.palette.primary.contrastText]
+    );
+
+    const displayAnalysisNotification = useCallback(
```

Comment:
Just picking a random spot - all these changes look good to me, but I don't think they belong in `CyVerseAppBar`.  It could be that we need some overarching component that acts like the DesktopPresenter to handle some of these more global events - all this logic and calling bootstrap for example. Or, I still haven't had time to read up on contexts so I'm not sure if that's something we could take advantage of here.  Not necessarily something to change now, but something we should keep in mind.

Reply:
As a first step, I moved `Notifications` and `CustomIntercom` into its own Component. Hopefully, we will know what kind of refactor is needed as we continue to build the codebase.